### ISSUE TITLE
Documentation: aniMotum/SSM processing/code, trajectory only

### ISSUE THREAD

**Issue by MathewBiddle** [ioos/marine_life_data_network#15]
### Who is requesting this?

@ioos/marine-life 

### What is being requested?

Documentation of what ATN is performing with respect to state space models (eg. foiegras/aniMotum/SSM processing/code). A description of what data are being ran through the process, what the processing entails (could be a link to an open source tool like aniMotum), how ATN configures the processing, and where that information goes (only in the portal, added to data files, metadata files, etc.). Enough information for someone unfamiliar with animal tracking to gain an understanding of how the data are being processed. 

Should probably be only 1/2 page in length, including some flow charts.

If possible, include a short code example of how data are processed. This could be a code snippet in the documentation or a Jupyter notebook published at https://ioos.github.io/ioos_code_lab/content/intro.html

### What is the requested deadline and why?

Initial publication of materials by 2026-05-01. To set a short turn deadline. 

### What is the current status quo (i.e., what happens if this does not get done)?

Marine life team will not understand nor be able to reference to what and how aniMotum/SSM processing is bring done in ATN.

### Provide a description or any other important information.

What indicates this is done?

Multiple actions could indicate this activity is done. The options are listed below but depend on what path the effort takes. These can be "either/or" but also "and" statements.

- [ ] A pull request to https://github.com/ioos/ioos-atn-data/ with documentation of the process.
- [ ] The documentation is published on GitHub at https://github.com/ioos/ioos-atn-data/

cc @mmckinzie

**Comment by mmckinzie** [comment]
What is being requested??

- (Axiom) Run all newly auto-ingested satellite telemetry data through aniMotum, use v. [1.1-04](https://ianjonsen.r-universe.dev/aniMotum), use R-Universe Package, package will not be available on Cran, additional details available here: [Fit Continuous-Time State-Space and Latent Variable Models for Quality Control of Argos Satellite (and Other) Telemetry Data and for Estimating Changes in Animal Movement • aniMotum (ianjonsen.github.io)](https://ianjonsen.github.io/aniMotum/index.html) and [ianjonsen/aniMotum: fit latent variable movement models to animal tracking data (github.com)](https://github.com/ianjonsen/aniMotum) 
- (Axiom) Release input and output files via the ADR
- (Axiom) Visualize output file in portal – projects pages, deployment pages, all and r/t layers (NOTE: Is it possible to make aniMotum outputs a visualization style, .i.e., hexes, tracks-raw, tracks-aniMotum?)
- (Axiom) Update portal language – update foie gras labels to aniMotum and location filter from unknown to aniMotum
- (Axiom) Work with ATN DC and code/package developer (Ian Jonsen) to troubleshoot errors/issues and improve upon base code to increase percentage of deployments which pass (currently at 70%) and improve outputted tracks, e.g., reduce angles, ensure outliers are successfully remove, including Z quality positions and positions outside of the PI reported deployment window 
- (Axiom) Retroactively run all deployments through aniMotum that were previously run through foie gras, update input/output files in ADR, update visualizations in portal
- (Axiom) Publish source code and processing steps on ATN GitHub, link to ML GitHub

Deadline: First three bullets already complete (integrating into portal as a visualization style would be a new task), fourth bullet – July 1, 2024, fifth bullet – September 1, 2024+, sixth and seventh bullets – November 30,2024

**Comment by laurabrenskelle** [comment]
This issue seems related to this Marine Life Data Network Trello card entitled: "1.8.4 add provenance statement to data catalog re: Foie Gras processing to help users understand what version of the data they are accessing". Is this correct, or is this task separate and needs its own new issue? Is 1.8.4 already complete? @mmckinzie @lmshowalter3

**Comment by mmckinzie** [comment]
@laurabrenskelle This task is related to the 1.8.4 and not it is not yet complete. We used to run data through Foie Gras and now we're using the updated aniMotum code, but we still need to add provenance statements to the ATN portal to provide context to how we're applying aniMotum and why. We have updated the portal labels from Foie Gras to aniMotum.  We may also still need to make some minor modifications to our processing steps to ensure a higher percentage of deployments 'pass' and to improve upon some of the output visualizations.  We also still need to retroactively run all developments through aniMotum, some older deployments have not been processed, while others have only been run through Foie Gras and not yet reprocessed through aniMotum.

**Comment by conniekot** [comment]
Still in progress: need to fix quality flags, unknown to anMotum

**Comment by conniekot** [comment]
@iwensu0313 Please update this issue with progress, if it has been incorporated into https://github.com/ioos/ioos_qc, and any links to the code - thanks!

**Comment by iwensu0313** [comment]
Megan already summarized remaining open items above. Here is a bulleted list with some additional information and action items that came out of our team's discussions:
- [ ] (Axiom) Update portal language – update location filter from unknown to aniMotum
    - Need to first update the data ingestion pipeline to include `aniMotum` in the "location class" column in the data tables that are processed by `aniMotum` (should be able to identify animotum processed files by looking for an appended `foiegras` in the file name, otherwise will have to re-run)
    - Then the front end can render this `aniMotum` value directly in the drop-down on the map instead of unknown
    - Some discussion about `aniMotum` not really being a location class, but a different data type altogether since they are modeled values, and thinking about other ways to represent it. For now, what is outlined above might be the easiest approach
- [ ] (Axiom) Work with ATN DC and code/package developer (Ian Jonsen) to troubleshoot errors/issues and improve upon base code to increase percentage of deployments which pass (currently at 70%) and improve outputted tracks, e.g., reduce angles, ensure outliers are successfully remove, including Z quality positions and positions outside of the PI reported deployment window
    - From Megan above: _We may also still need to make some minor modifications to our processing steps to ensure a higher percentage of deployments 'pass' and to improve upon some of the output visualizations._
- [ ] (Axiom) Retroactively run all deployments through aniMotum that were previously run through foie gras, update input/output files in ADR, update visualizations in portal
    - I think we are waiting on finalization of the first two items before doing this, is that right?
- [ ] (Axiom) Publish source code and processing steps on ATN GitHub, link to ML GitHub
    - Would be helpful to talk about what kind of information would be most helpful
    - The source code for aniMotum is here: https://github.com/ianjonsen/aniMotum
    - Configuration that we use for the aniMotum are these as recommended by Ian Jonsen
        - `model = "rw"` (random walk)
        - `time.step = 2"`
        - `grab.fit_type` is `predicted`
        - `grab.as_sf` is `False`
    - Regarding version:`install.packages("aniMotum", repos = c("https://cloud.r-project.org", "https://ianjonsen.r-universe.dev"), dependencies = TRUE)`
    - Are we looking for a contextual summary 1 pager situation describing the above?
- [ ] From Megan: _we still need to add provenance statements to the ATN portal to provide context to how we're applying aniMotum and why_
    - Do we have text drafted for this?

Note: We discussed at Nov Marine Life Program Monthly meeting to pause on new development work until March.

**Comment by MathewBiddle** [comment]
Reviewing this now. It looks like the scope of this issue expanded. The scope was initially documentation and it sounds like it's expanded to implementation. 

Additionally, from my understanding ATN isn't implementing any of this right now. https://github.com/ioos/marine_life_data_network/issues/162#issuecomment-3843120255

**Comment by MathewBiddle** [comment]
Holy smokes, this issue diverged. The sole purpose of this issue was to document what is or isn't happening in ATN with respect to SSM processing (foiegras, aniMotum, whatever else is happening).

Please narrow this topic back down to the **documentation** of what processing is actually happening. Add that documentation as a subpage to https://github.com/ioos/ioos-atn-data/.

The tasking of implementation/development/visualization for `aniMotum`/`foiegras`/other SSM should be tracked in another issue where we can evaluate the effort against current priorities. 

Moving this to ToDo.

**Comment by MathewBiddle** [comment]
@iwensu0313 I have updated the description https://github.com/ioos/marine_life_data_network/issues/15#issue-2253906412

**Comment by MathewBiddle** [comment]
closed by https://github.com/ioos/ioos-atn-data/pull/111

**Comment by MathewBiddle** [comment]
See https://ioos.github.io/ioos-atn-data/animotum.html
